### PR TITLE
Implemented Modal Component

### PR DESCRIPTION
**Pull request by nikilrselvam**
Two Main Modal Types: in-place and entire-screen. Type is specified as a prop 'type' to the Modal Component.

### CODE REVIEW

File: src/components/Modal/style.scss [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+.inline-modal-wrapper{
+    position: fixed;
+    height: auto;
+    width: auto;
+    z-index: 1000;
+    overflow: auto;
+    text-align: center;
+    transform: translate(-50%, -50%);
```
Comment (ycyvonne) [inline src/components/Modal/style.scss:65]: The `transform` property actually needs additional support for other browsers, can you add some vendor prefixes? 
Example:

```
 -webkit-transform: rotate(30deg); /* Ch <36, Saf 5.1+, iOS < 9.2, An =<4.4.4 */
      -ms-transform: rotate(30deg); /* IE 9 */
          transform: rotate(30deg); /* IE 10, Fx 16+, Op 12.1+ */
```

File: src/components/Page/Page.jsx [hunk 2]
```diff
@@ -5,9 +5,6 @@ import NavBar from '../../components/NavBar/NavBar';
 // The Page Component appropriately renders the NavBar and any children components
 
 class Page extends Component {
-    constructor(props) {
```
Comment (ycyvonne) [inline src/components/Page/Page.jsx:8]: can you put this back in? since you access `this.props` in the file

File: src/components/Modal/Modal.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        this.setState({ open: false },() => {
+            document.removeEventListener('click', this.handleClick);
+          });
+        document.body.classList.remove('modal-visible');
+        this.props.closeModal();
+    }
+
+
+    handleClick= (e)=> {
+    // if (e.target.className === "modal-wrapper") {
```
Comment (ycyvonne) [inline src/components/Modal/Modal.jsx]: can you clean up this comment?

File: src/components/Modal/Modal.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        }
+
+        this.close();
+    }
+
+    render() {
+
+        if(this.state.open)
+            this.show();
+        const mdl=this.state.type==='entire-screen'?(
```
Comment (ycyvonne) [inline src/components/Modal/Modal.jsx]: is there a reason why you store this into a const variable instead of returning directly?

**Review by ycyvonne — CHANGES_REQUESTED** [review]
Looks good! I just had some nitpicks.

File: src/components/Modal/Modal.jsx (hunk 4 above)
Comment (nikilrselvam) [reply src/components/Modal/Modal.jsx]: Nope! It was just trying to make it clearer for me personally! I've changed it to a single return statement now!

File: src/components/Modal/style.scss (hunk 1 above)
Comment (nikilrselvam) [reply src/components/Modal/style.scss:65]: Wow I had no idea something like this existed. I just read up on it now! Thanks!

**Pull request merged**